config/helpers/bootstrap.go

```
@@ -0,0 +1,193 @@
+type TransformFileFunc func(bs []byte) ([]byte, error)
+
+// Option allows to customize the bootstrap process.
+type Option struct {
+	// TransformFileFunc is a function that transforms a resource file before being applied to the cluster.
+	TransformFile TransformFileFunc
+}
+
+// ReplaceOption allows to customize the bootstrap process.
+func ReplaceOption(pairs ...string) Option {
+	return Option{
+		TransformFile: func(bs []byte) ([]byte, error) {
+			if len(pairs)%2 != 0 {
+				return nil, fmt.Errorf("odd number of arguments: %v", pairs)
+			}
+			for i := 0; i < len(pairs); i += 2 {
+				bs = bytes.ReplaceAll(bs, []byte(pairs[i]), []byte(pairs[i+1]))
+			}
+			return bs, nil
+		},
+	}
+}
+
+// Bootstrap creates a list of CRDs and then the resources in a package's fs by
+// continuously retrying the list. This is blocking, i.e. it only returns (with error)
+// when the context is closed or with nil when the bootstrapping is successfully completed.
+func Bootstrap(ctx context.Context, crdClient apiextensionsclient.Interface, dynamicClient dynamic.Interface, fs embed.FS, crds []metav1.GroupResource, opts ...Option) error {
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(crdClient.Discovery()))
+
+	if err := wait.PollImmediateInfiniteWithContext(ctx, time.Second, func(ctx context.Context) (bool, error) {
+		if err := configcrds.Create(ctx, crdClient.ApiextensionsV1().CustomResourceDefinitions(), crds...); err != nil {
+			klog.Errorf("failed to bootstrap CRDs: %v", err)
+			return false, nil // keep retrying
+		}
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("failed to bootstrap CRDs into root workspace: %w", err)
+	}
+
+	// bootstrap non-crd resources
+	var transformers []TransformFileFunc
+	for _, opt := range opts {
+		transformers = append(transformers, opt.TransformFile)
+	}
+	return wait.PollImmediateInfiniteWithContext(ctx, time.Second, func(ctx context.Context) (bool, error) {
+		// +lint:noerrcheck
```

(No action required) Why is this required here but not above line 77?

---

I guess this is for line 95, ignoring the error